clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -77,6 +76,7 @@ public void testPartitioner() throws Exception {
         assertEquals("Partition should be correct", 1, metadata.get().partition());
         producer.clear();
         assertEquals("Clear should erase our history", 0, producer.history().size());
+        producer.close();
```

Comment:
For this PR I tried to use try-with-resource pattern instead of an existing try-catch block, when possible. I'm not sure about the best-practice here, but my feeling is the code is more readable this way, plus I'm not sure if try-with-resource adds much value here. Please advise if you disagree.